api/src/services/generate-xform.js

```diff
@@ -70,33 +70,90 @@ const transform = (formXml, stylesheet) => {
   });
 };
 
-const removeLast = (haystack, needle) => {
-  const index = haystack.lastIndexOf(needle);
-  if (index === -1) {
-    return haystack;
-  }
-  return haystack.slice(0, index) + haystack.slice(index + needle.length);
+const replaceMarkdown = html => {
+  return html
+    // headings
+    .replace(/\n# (.*)\n/gm, '<h1>$1</h1>')
+    .replace(/\n## (.*)\n/gm, '<h2>$1</h2>')
+    .replace(/\n### (.*)\n/gm, '<h3>$1</h3>')
+    .replace(/\n#### (.*)\n/gm, '<h4>$1</h4>')
+    .replace(/\n##### (.*)\n/gm, '<h5>$1</h5>')
+
+    // font styles
+    .replace(/__([^\s]([^_]*[^\s])?)__/gm, '<strong>$1</strong>')
+    .replace(/\*\*([^\s]([^*]*[^\s])?)\*\*/gm, '<strong>$1</strong>')
+    .replace(/\s_([^_\s]([^_]*[^_\s])?)_/gm, ' <em>$1</em>')
+    .replace(/\*([^*\s]([^*]*[^*\s])?)\*/gm, '<em>$1</em>')
+
+    // urls containing tags
+    .replace(
+      /\[([^\]]*)\]\(([^)]*<[^>]*>[^)]*)\)/gm,
+      '<a href="#" target="_blank" rel="noopener noreferrer" class="dynamic-url">' +
+      '$1<span class="url hidden">$2</span>' +
+      '</a>'
+    )
+
+    // plain urls
+    .replace(/\[([^\]]*)\]\(([^)]+)\)/gm, '<a href="$2" target="_blank" rel="noopener noreferrer">$1</a>')
+
+    // new lines
+    .replace(/\n/gm, '<br />')
+
+    // convert embedded html
+    .replace(/&lt;([\s\S]*?)&gt;/gm, '<$1>')
+    .replace(/&amp;/g, '&')
+    .replace(/&quot;/g, '"')
+    .replace(/&#039;/g, '\'');
 };
 
-const removeRootNode = (string, node) => {
-  return removeLast(string.replace(node, ''), ROOT_CLOSE);
+// inspired by enketo/enketo-transformer
+const replaceAllMarkdown = formString => {
+
+  const form = htmlParser.parse(formString).querySelector('form');
+  const replacements = {};
+
+  const questions = form.querySelectorAll('span.question-label');
```

Comment:
I double-checked with a report out on https://getodk.org/xlsform/ using just Enketo code and verified that the new enketo-transform code supports markdown formatting in places like question labels (and not just `.note` elements). So, I think that going forwards this is probably functionality we also want to support.

I briefly talked to @garethbowen about this. He is aware that this is new behavior for the CHT and that is has the potential for unintentional consequences, but essentially felt that we could mitigate the risks.  I am planning on adding this to the doc for this PR and making sure it ends up in some kind of release communication directed at app developers so that they are aware of the new functionality and the possible effects on existing formatting.  @dianabarsan let me know if you have additional concerns here that need to be discussed.